File: QLBHDIENTU/src/mypack/SanPham.java
package mypack;

public class SanPham {
	int idsanpham;
	String tensanpham;
	double dongia;
	int thoihanbaohanh;
	int namsanxuat;
	NhaCungCap NhaCungCap;
	LoaiHang LoaiHang;

	public SanPham (SanPham sp){
		this.idsanpham = sp.idsanpham;
		this.tensanpham = sp.tensanpham;
		this.dongia = sp.dongia;
		this.thoihanbaohanh = sp.thoihanbaohanh;
		this.namsanxuat = sp.namsanxuat;
		this.NhaCungCap = sp.NhaCungCap;
		this.LoaiHang = sp.LoaiHang;
	}
	
	public SanPham (int idsanpham, String tensanpham, double dongia, int thoihanbaohanh,
			int namsanxuat, NhaCungCap NhaCungCap, LoaiHang LoaiHang){
		this.idsanpham = idsanpham;
		this.tensanpham = tensanpham;
		this.dongia = dongia;
		this.thoihanbaohanh = thoihanbaohanh;
		this.namsanxuat = namsanxuat;
		this.NhaCungCap =  new NhaCungCap(NhaCungCap);
		this.LoaiHang = LoaiHang;
	}
	public int getIDSanPham() {
		return idsanpham;
	}
	
	public String getTenSanPham() {
		return tensanpham;
	}
	
	public double getDonGia() {
		return dongia;
	}
	
	public int getThoiHanBaoHanh() {
		return thoihanbaohanh;
	}
	
	public int getNamSanXuat() {
		return namsanxuat;
	}
	
	public NhaCungCap getNhaCungCap() {
		return NhaCungCap;
	}
	
	public LoaiHang getLoaiHang() {
		return LoaiHang;
	}
	
}

File: QLCHDoDienTu-master/Model/SanPham.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLCHDoDienTu {
	public class SanPham {
		public int IDSanPham { get; set; }
		public string TenSanPham { get; set; }
		public decimal DonGia { get; set; }
		public int ThoiHanBaoHanh { get; set; }
		public int NamSanXuat { get; set; }
		public NhaCungCap NhaCungCap { get; set; }
		public LoaiHang LoaiHang { get; set; }

		public SanPham(SanPham sp)
		{
			this.IDSanPham = sp.IDSanPham;
			this.TenSanPham = sp.TenSanPham;
			this.DonGia = sp.DonGia;
			this.ThoiHanBaoHanh = sp.ThoiHanBaoHanh;
			this.NamSanXuat = sp.NamSanXuat;
			this.NhaCungCap = sp.NhaCungCap;
			this.LoaiHang = sp.LoaiHang;
		}
		public SanPham(int IDSanPham_, string TenSanPham_, decimal DonGia_, int ThoiHanBaoHanh_, int NamSanXuat_, NhaCungCap NhaCungCap_, LoaiHang LoaiHang_) {
			this.IDSanPham = IDSanPham_;
			this.TenSanPham = TenSanPham_;
			this.DonGia = DonGia_;
			this.ThoiHanBaoHanh = ThoiHanBaoHanh_;
			this.NamSanXuat = NamSanXuat_;
			this.NhaCungCap = NhaCungCap_;
			this.LoaiHang = LoaiHang_;
		}
	}
}
